### DISCUSSION TITLE
mpisppy: using model from PySP and using hubs and spokes on top

### DISCUSSION THREAD

**Discussion by abodh** [Pyomo/mpi-sppy#284] [Q&A]
Hello, 

Is there an example in mpisppy where we can import a concrete model from PySP with networkx scenario creator and invoke hubs and spoke cylinders to run the PySP imported model in parallel? I want to use Progressive Hedging to do this. 

I just started reviewing mpisppy and would appreciate any help on this :) Furthermore, are there good examples of running two-stage problems in parallel using mpisppy without learning too many HPC terminologies?

Regards,
Abodh

**Comment by DLWoodruff** [comment]
Hi Abodh,
  When you say "networkx scenario creator" I think you mean networkx
scenario tree creator. For two-stage problems in mpi-sppy, tree creation is
extremely easy. The scenario creator just has a function call to
sputils.attach_root_node like the one in the farmer.py example:

    varlist = [model.DevotedAcreage]
    sputils.attach_root_node(model, model.FirstStageCost, varlist)

The farmer example is a pretty good example. You don't have to learn too
much HPC terminology, but you do have to install MPI and mpi4py as
described in the github readme and the documentation:
https://mpi-sppy.readthedocs.io/en/latest/

**Reply by abodh** [reply]
I count your response as a boost to my learning experiences. I will check the farmer example as you suggested and come back here if I have more questions. Thank you, Dr. Woodruff.
